### PR TITLE
Improve and extend changelog logs

### PR DESCRIPTION
**Pull request by felixfontein**
##### SUMMARY
The current changelog docs have some problems:
1. Collection maintainers sometimes only know the docs starting at `Creating a changelog fragment` and don't realize that new modules and (most) plugins don't need a fragment.
2. Add a reference early-on of `Creating a changelog fragment` to the next section with the more precise format of changelog fragments. The next section is quite often ignored (in c.g I have to add at least one suggestion to every changelog fragment that hasn't been created by a PR author who already had several other PRs merged).
3. Adds the information from https://github.com/ansible-community/antsibull-changelog/blob/main/docs/changelogs.rst#adding-new-roles-playbooks-test-and-filter-plugins as a new section `Changelog fragment entry format for new jinja2 plugins, roles and playbooks`.

CC @Andersson007 

##### ISSUE TYPE
- Docs Pull Request

##### COMPONENT NAME
docs/docsite/rst/community/development_process.rst

### CODE REVIEW

File: docs/docsite/rst/community/development_process.rst
Comment (Andersson007) [inline docs/docsite/rst/community/development_process.rst]:
```suggestion
Changelogs help users and developers keep up with changes to ansible-core and Ansible collections. Ansible and many collections build changelogs for each release from fragments. For ansible-core and collections using this model, you **must** add a changelog fragment to any PR that changes functionality or fixes a bug.

You do not need a changelog fragment for PRs that:

* add new modules and plugins, because our tooling does that for you automatically;
* contain only documentation changes;

.. note::
  Some collections require a changelog fragment for every pull request. They use the ``trivial:`` section for entries mentioned above that will be skipped when building a release changelog.

```
Should we get rid of the period before `because`?

Also `Changelogs help users and developers keep up with changes to Ansible and Ansible collections.`
I'd change this to:
```
Changelogs help users and developers keep up with changes to Ansible.
```
because Ansible collections are part of Ansible
or, if ansible-core is implied, to
```
Changelogs help users and developers keep up with changes to Ansible-core and Ansible collections.
```
and in the next sentence too
(not sure if it should be Ansible-core, Ansible-Core, or ansible-core)

Also FYI: I'll replace all PRs to "pull requests" in another PR later, it's too many places to fix now.

File: docs/docsite/rst/community/development_process.rst
Comment (Andersson007) [inline docs/docsite/rst/community/development_process.rst]:
```suggestion
Changelog fragment entry format for new jinja2 plugins, roles, and playbooks
```

File: docs/docsite/rst/community/development_process.rst
Comment (Andersson007) [inline docs/docsite/rst/community/development_process.rst]:
```suggestion
While new modules and plugins that are not jinja2 filter or test plugins are mentioned automatically in the generated changelog, jinja2 filter and test plugins, roles, and playbooks are not. To make sure they are mentioned, a changelog fragment in a specific format is needed:
```

File: docs/docsite/rst/community/development_process.rst
Comment (felixfontein) [inline docs/docsite/rst/community/development_process.rst]:
```suggestion
^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
```

File: docs/docsite/rst/community/development_process.rst
Comment (Andersson007) [reply docs/docsite/rst/community/development_process.rst]: @felixfontein your thoughts ^

Comment (felixfontein) [reply docs/docsite/rst/community/development_process.rst]: Can you adjust your suggestion so that it uses `;`/`.` for the list? Then I'd merge it.

About the second part, I'd rather use `Changelogs help users and developers keep up with changes to ansible-core and Ansible collections.` (the correct spelling is `ansible-core` with lower-case `a`). Also not all collections are part of Ansible :)

Comment (Andersson007) [reply docs/docsite/rst/community/development_process.rst]: Done, PTAL. If it's not what you meant, feel free to change it yourself, np

Comment (Andersson007) [reply docs/docsite/rst/community/development_process.rst]: After that, could you please rephrase the second and the third sentences: 1) it starts with Ansible and 2) there are, IMO, too many repetitions in the first paragraph

File: docs/docsite/rst/community/development_process.rst
Comment (felixfontein) [inline docs/docsite/rst/community/development_process.rst:146]: @Andersson007 I don't think it's a good idea to remove any use of Ansible/ansible here, as it will destroy the meaning. Maybe ok would be:
```suggestion
Changelogs help users and developers keep up with changes to ansible-core and Ansible collections. Ansible and many collections build changelogs for each release from fragments. For them, you **must** add a changelog fragment to any PR that changes functionality or fixes a bug.
```
but even that doesn't sound right to me.

Comment (Andersson007) [reply docs/docsite/rst/community/development_process.rst:146]: we still have `.. Ansible and many collections..`. If we're talking about ansible-core it should be `ansible-core` but if we start with it, it should start with a capital letter:) Can we rephrase the sentence to put `ansible-core`, say, in the middle of the sentence?

File: docs/docsite/rst/community/development_process.rst
Comment (Akasurde) [inline docs/docsite/rst/community/development_process.rst]:
```suggestion
* add new modules and plugins, because Ansible tooling does that automatically;
```

**Comment by sivel** [comment]
This will need review by the core team as well, and not just docs.  I cannot commit to adhering to these changes until they are reviewed and potentially adjusted to meet our workflow.  We won't necessarily adjust our workflow for the sake of docs.

Note: I haven't looked at this at all, and while I am adding myself for review, I may not be the reviewer.  Just adding myself to indicate that core needs to review.

**Comment by felixfontein** [comment]
@sivel the changes in this PR should not really affect core team's workflow, most of the things that are now explained in more detail is what you already do. The only new part is the changelog entry format for new jinja2 plugins, roles and playbooks, which didn't exist until recently.

**Comment by sivel** [comment]
Understood, my statements were just trying to convey we haven't read it, and want to make sure we review.

**Review by sivel — APPROVED** [review]

**Review by samccann — APPROVED** [review]

**Comment by felixfontein** [comment]
@Andersson007 @samccann  @Akasurde @sivel thanks for reviewing and merging!

File: docs/docsite/rst/community/development_process.rst
Comment (webknjaz) [inline docs/docsite/rst/community/development_process.rst:161]: @felixfontein FYI https://jakubmarian.com/how-vs-what-does-it-look-like-in-english/ / https://new.testmagic.com/sat-questions/what-vs-how/

**Pull request merged**
